incydr/types.py

```diff
@@ -0,0 +1,38 @@
+import click
+
+
+# TODO: do we want this ?
+# class AutoDecodedFile(click.File):
+#     """Attempts to autodetect file's encoding prior to normal click.File processing."""
+#
+#     def convert(self, value, param, ctx):
+#         try:
+#             with open(value, "rb") as file:
+#                 self.encoding = chardet.detect(file.read())["encoding"]
+#             if self.encoding is None:
+#                 # TODO: logging
+#                 click.echo(f"Failed to detect encoding of file: {value}")
+#         except Exception:
+#             pass  # we'll let click.File do it's own exception handling for the filepath
+#
+#         return super().convert(value, param, ctx)
+
+
+class FileOrString(click.Path):
+    """Declares a parameter to be a file (if the argument begins with `@`), otherwise accepts it as
```

Comment:
Ope, nevermind.. I see what you mean.

Reply:
we could just only allow one of these options at a time, as this will also get more complicated with the addition of JSONLines file format as an option, as it would be difficult to resolve `--format <csv|json-lines>` to multiple input files too.